### PR TITLE
Add builder for Git

### PR DESCRIPTION
**Pull request by giordano**
It doesn't work on macOS, as it needs curl (which is usually an optional dependency, but would be useful to have everywhere anyway), and on Windows, where it fails to configure because of missing `socklen_t` or equivalent.

A possibility for Windows could be to simply use the pre-built `.exe`s in https://github.com/git-for-windows/git/releases/tag/v2.23.0.windows.1 if we're sure that they're self-contained and don't need other dynamic libraries.  As an alternative, we can take the full `.tar.bz2` tarballs and install the `/usr` and `bin` directories in `${prefix}`

CC: @DilumAluthge

### CODE REVIEW

**Comment by giordano** [comment]
I couldn't do a single script for all platforms because the git-for-windows is a tarbomb that would explode in `$WORKSPACE/srcdir`, and would also be a conflict between the 32- and the 64-bit builds, even though we only care about the mingw* directories.  In addition, these Windows tarbombs are self-contained (they already ship all dependencies), so for Windows we really don’t want to have dependencies.  In the end, `build_tarballs` needs platform-dependent sources and dependencies.

Note: I couldn't create the tarball for 32-bit Windows, auditor failed:
```
Info: Checking libexec/git-core/msobj140.dll with RPath list String[]
Warning: Linked library De could not be resolved and could not be auto-mapped
Warning: Linked library 4 could not be resolved and could not be auto-mapped
Warning: Linked library GetCurrentThreadId could not be resolved and could not be auto-mapped
Warning: Linked library P could not be resolved and could not be auto-mapped
ERROR: LoadError: Base.InvalidCharError{Char}('\xd0')
Stacktrace:
 [1] invalid_char(::Char) at ./char.jl:85
 [2] UInt32(::Char) at ./char.jl:130
 [3] convert at ./char.jl:180 [inlined]
 [4] cconvert at ./essentials.jl:387 [inlined]
 [5] lowercase(::Char) at ./strings/unicode.jl:245
 [6] map(::typeof(lowercase), ::String) at ./strings/basic.jl:572
 [7] lowercase at ./strings/unicode.jl:531 [inlined]
 [8] should_ignore_lib(::String, ::ObjectFile.COFF.COFFHandle{IOStream}) at /home/mose/.julia/dev/BinaryBuilder/src/auditor/dynamic_linkage.jl:193
 [9] #check_dynamic_linkage#128(::Base.TTY, ::Windows, ::Bool, ::Bool, ::Bool, ::typeof(BinaryBuilder.check_dynamic_linkage), ::ObjectFile.COFF.COFFHandle{IOStream}, ::Prefix, ::Array{String,1}) at /home/mose/.julia/dev/BinaryBuilder/src/Auditor.jl:265
 [10] (::BinaryBuilder.var"#kw##check_dynamic_linkage")(::NamedTuple{(:io, :platform, :silent, :verbose, :autofix),Tuple{Base.TTY,Windows,Bool,Bool,Bool}}, ::typeof(BinaryBuilder.check_dynamic_linkage), ::ObjectFile.COFF.COFFHandle{IOStream}, ::Prefix, ::Array{String,1}) at /home/mose/.julia/dev/BinaryBuilder/src/Auditor.jl:249
 [11] (::BinaryBuilder.var"##114#121"{Base.TTY,Windows,Bool,Bool,Prefix,Array{String,1},String})(::ObjectFile.COFF.COFFHandle{IOStream}) at /home/mose/.julia/dev/BinaryBuilder/src/Auditor.jl:76
 [12] readmeta(::BinaryBuilder.var"##114#121"{Base.TTY,Windows,Bool,Bool,Prefix,Array{String,1},String}, ::String) at /home/mose/.julia/packages/ObjectFile/UYvnP/src/Abstract/ObjectHandle.jl:158
 [13] #audit#112(::Base.TTY, ::Windows, ::Bool, ::Bool, ::Bool, ::typeof(audit), ::Prefix) at /home/mose/.julia/dev/BinaryBuilder/src/Auditor.jl:61
 [14] (::BinaryBuilder.var"#kw##audit")(::NamedTuple{(:platform, :verbose, :autofix),Tuple{Windows,Bool,Bool}}, ::typeof(audit), ::Prefix) at /home/mose/.julia/dev/BinaryBuilder/src/Auditor.jl:36
 [15] (::BinaryBuilder.var"##292#301"{Bool,Bool,Bool,Bool,Bool,Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},String,String,VersionNumber,String,Array{Windows,1},Array{ExecutableProduct,1},Array{Any,1},Dict{Any,Any}})(::String) at /home/mose/.julia/dev/BinaryBuilder/src/AutoBuild.jl:619
 [16] #mktempdir#20(::String, ::typeof(mktempdir), ::BinaryBuilder.var"##292#301"{Bool,Bool,Bool,Bool,Bool,Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},String,String,VersionNumber,String,Array{Windows,1},Array{ExecutableProduct,1},Array{Any,1},Dict{Any,Any}}, ::String) at ./file.jl:656
 [17] mktempdir at ./file.jl:654 [inlined] (repeats 2 times)
 [18] #autobuild#290(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::Nothing, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(autobuild), ::String, ::String, ::VersionNumber, ::Array{Pair{String,String},1}, ::String, ::Array{Windows,1}, ::Array{ExecutableProduct,1}, ::Array{Any,1}) at /home/mose/.julia/dev/BinaryBuilder/src/AutoBuild.jl:414
 [19] (::BinaryBuilder.var"#kw##autobuild")(::NamedTuple{(:verbose, :debug),Tuple{Bool,Bool}}, ::typeof(autobuild), ::String, ::String, ::VersionNumber, ::Array{Pair{String,String},1}, ::String, ::Array{Windows,1}, ::Array{ExecutableProduct,1}, ::Array{Any,1}) at /home/mose/.julia/dev/BinaryBuilder/src/AutoBuild.jl:398
 [20] #build_tarballs#275(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(build_tarballs), ::Array{String,1}, ::String, ::VersionNumber, ::Array{Pair{String,String},1}, ::String, ::Array{Windows,1}, ::Array{ExecutableProduct,1}, ::Array{Any,1}) at /home/mose/.julia/dev/BinaryBuilder/src/AutoBuild.jl:157
 [21] build_tarballs(::Array{String,1}, ::String, ::VersionNumber, ::Array{Pair{String,String},1}, ::String, ::Array{Windows,1}, ::Array{ExecutableProduct,1}, ::Array{Any,1}) at /home/mose/.julia/dev/BinaryBuilder/src/AutoBuild.jl:23
 [22] top-level scope at /home/mose/repo/builders/Yggdrasil/G/Git/build_tarballs.jl:83
 [23] include(::Module, ::String) at ./Base.jl:376
 [24] exec_options(::Base.JLOptions) at ./client.jl:297
 [25] _start() at ./client.jl:493
in expression starting at /home/mose/repo/builders/Yggdrasil/G/Git/build_tarballs.jl:80
```
No problems with 64-bit Windows

**Comment by giordano** [comment]
Also, of course the way I organised this script breaks passing platforms as arguments, I _strongly_ recommend **not** doing that :sweat_smile:

File: G/Git/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+# Bash recipe for building across all Unices
+script_unix = raw"""
+cd $WORKSPACE/srcdir/git-*/
+
+# We need a native "msgfmt" to cross-compile
+apk add tcl gettext
+
+# Git doesn't want to be cross-compiled, but we swear that we're not going to
+# cross-compile
```
Comment (staticfloat) [inline G/Git/build_tarballs.jl]: 😂

**Comment by staticfloat** [comment]
Congrats with the error on win32 BTW; you've found a bug in ObjectFile that I still don't understand. :P

**Pull request merged**